src/entities/Updates/routes.ts

```diff
@@ -158,6 +159,51 @@ async function updateProposalUpdate(req: WithAuth<Request<{ proposal: string }>>
   return true
 }
 
+async function deleteProposalUpdate(req: WithAuth<Request<{ proposal: string }>>) {
+  const { id } = req.body
+  const update = await UpdateModel.findOne(id)
+  const proposalId = req.params.proposal
+
+  if (!update) {
+    throw new RequestError(`Update not found: "${id}"`, RequestError.NotFound)
+  }
+
+  if (!update?.completion_date) {
+    throw new RequestError(`Update is not completed: "${update.id}"`, RequestError.BadRequest)
+  }
+
+  const user = req.auth
+  const proposal = await ProposalModel.findOne<ProposalAttributes>({ id: req.params.proposal })
+
+  const isAuthorOrCoauthor =
+    user && (proposal?.user === user || (await isCoauthor(proposalId, user))) && update.author === user
```

Comment:
Only the authors of the update should be able to delete? why not the other co-authors or the author of the proposal?